BaragonUI/app/components/requestDetail/RequestDetail.jsx

```diff
@@ -1,23 +1,60 @@
-import React, { Component, PropTypes } from 'react';
+import React, { PropTypes } from 'react';
 import { connect } from 'react-redux';
-import classNames from 'classnames';
-import { Link } from 'react-router';
 import rootComponent from '../../rootComponent';
-import { refresh } from '../../actions/ui/serviceDetail';
+import { refresh } from '../../actions/ui/requestDetail';
 import Utils from '../../utils';
 
-class RequestDetail extends Component {
-  render() {
-    // TODO - render out info nicely + link to service page/group page
-    return <h1>Request {this.props.params.requestId}</h1>
-  }
+import JSONButton from '../common/JSONButton';
 
-  static propTypes = {
-    params: PropTypes.object.isRequired,
-    response: React.PropTypes.object
-  };
-}
+import SummaryPanel from './SummaryPanel';
+import AgentResponsesPanel from './AgentResponsesPanel';
+import OriginalRequestPanel from './OriginalRequestPanel';
 
-export default connect((state, ownProps) => ({
+const RequestDetail = ({requestId, serviceId, agentResponses, message, request, response}) => {
+  return (
+    <div>
+      <div className="row detail-header">
+        <div className="col-md-10">
+          <h3>Request: {requestId}</h3>
+        </div>
+        <div className="col-md-2 button-container">
+          <JSONButton object={response} showOverlay={true}>
+            <a className="btn btn-default">JSON</a>
+          </JSONButton>
+        </div>
+      </div>
+      <div className="row">
+        <SummaryPanel
+          serviceId={serviceId}
+          message={message}
+        />
+      </div>
+      <div className="row">
+        <AgentResponsesPanel agentResponses={agentResponses} />
+      </div>
+      <div className="row">
+        <OriginalRequestPanel request={request} />
+      </div>
+    </div>
+  );
+};
+
+RequestDetail.propTypes = {
+  requestId: PropTypes.string,
+  serviceId: PropTypes.string,
+  agentResponses: PropTypes.object,
+  message: PropTypes.string,
+  request: PropTypes.object,
+  response: PropTypes.object,
```

Comment:
Since we know the form response takes, we can actually be more detailed here using `PropTypes.shape`

e.g.
```
response: PropTypes.shape({
  loadBalancerRequestId: PropTypes.string,
  agentResponses: PropTypes.arrayOf(PropTypes.shape({ ... 
...
```